### DISCUSSION TITLE
Feature Request: Use F2 key to rename folder

### DISCUSSION THREAD

**Discussion by pcgeek86** [massCodeIO/massCode#205] [Ideas]
## Summary

The F2 key is a de facto standard for renaming items in user interfaces.

Source: https://www.google.com/search?q=f2+common+shortcut

## Expected Result

When a folder is selected, in the left-hand navigation pane, and I press the F2 key, I should be able to rename the folder.

## Actual Result

Nothing happens when I have a folder selected, and press F2.

**Comment by antonreshetov** [comment]
It's strange to expect anything )) Try for example to do the same in VS Code, Notion, etc., you'll be surprised that your expectations are not met.

**Reply by pcgeek86** [reply]
@antonreshetov In VSCode, if I selected a file/folder in the **Explorer** pane, and press `F2`, it invokes a rename operation.

Anyway, it's just a feature request to think about for sometime in the future.

**Reply by antonreshetov** [reply]
I do not, apparently it's because of the Atom Keymap extension.

Tip: You can double-click a folder to rename it. The same works for fragments.

**Reply by pcgeek86** [reply]
Gotcha, I was just trying to provide a suggestion to make massCode more user-friendly. It's very common to use F2 to rename things in various applications, so thought maybe massCode could follow the same convention.

If you don't want to implement it, and no one else has time to contribute it, then that's okay. I would contribute it, but I'm not primarily a JavaScript developer.

**Reply by antonreshetov** [reply]
Thanks for the suggestion, I think I'll add it in the future. It's just the wording itself confused me "expecting something".

**Reply by pcgeek86** [reply]
@antonreshetov Oh okay, I just use that terminology because it's testing verbiage for submitting bug reports and feature requests. It helps the developer to understand how someone would expect a feature to work, even though it has not been implemented yet. Thanks for your consideration.

I just discovered massCode a few minutes ago, and am playing around with it, and love it the concept! Thank you for creating it. Cheers
